parcels/particleset.py

```diff
@@ -823,21 +809,25 @@ def execute(self, pyfunc=AdvectionRK4, pyfunc_inter=None, endtime=None, runtime=
             Length of the timestepping loop. Use instead of endtime.
             It is either a timedelta object or a positive double. (Default value = None)
         dt :
-            Timestep interval to be passed to the kernel.
+            Timestep interval (in seconds) to be passed to the kernel.
             It is either a timedelta object or a double.
-            Use a negative value for a backward-in-time simulation. (Default value = 1.)
+            Use a negative value for a backward-in-time simulation. (Default value = 1 second)
         output_file :
             mod:`parcels.particlefile.ParticleFile` object for particle output (Default value = None)
         verbose_progress : bool
             Boolean for providing a progress bar for the kernel execution loop. (Default value = True)
         postIterationCallbacks :
-            Optional) Array of functions that are to be called after each iteration (post-process, non-Kernel) (Default value = None)
+            Optional, array of functions that are to be called after each iteration (post-process, non-Kernel) (Default value = None)
         callbackdt :
             Optional, in conjecture with 'postIterationCallbacks', timestep interval to (latest) interrupt the running kernel and invoke post-iteration callbacks from 'postIterationCallbacks' (Default value = None)
         pyfunc_inter :
             (Default value = None)
         delete_cfiles : bool
             Whether to delete the C-files after compilation in JIT mode (default is True)
+
+        Notes
+        -----
+        Parcels interoperates between Python and (in the case of JIT mode) compiled C code. ParticleSet.execute() acts as the main entrypoint for simulations, and provides the simulation time-loop. This method encapsulates logic deciding when control flow is handled to C (for kernel execution), and Python (for writing output files, reading in fields for new timesteps, for adding new particles to the simulation domain, for stopping the simulation, and for executing custom functions `postIterationCallbacks` provided by the user).
```

Comment:
```suggestion
        ParticleSet.execute() acts as the main entrypoint for simulations, and provides the simulation time-loop. This method encapsulates the logic controlling the switching between kernel execution (where control in handed to C in JIT mode), output file writing, reading in fields for new timesteps, adding new particles to the simulation domain, stopping the simulation, and executing custom functions (`postIterationCallbacks` provided by the user).
```

Reply:
Looks good! Updated the docstring to also use RST formatting.